src/zkbob/manager/AllowListOperatorManager.sol

```diff
@@ -0,0 +1,132 @@
+    // if false, anyone can be an operator
+    bool public allowListEnabled;
+
+    // mapping of whitelisted operator addresses
+    mapping(address => bool) public operators;
+
+    // mapping of fee receivers for operators
+    mapping(address => address) public operatorFeeReceiver;
+
+    event UpdateOperator(address indexed operator, address feeReceiver, bool allowed);
+    event UpdateAllowListEnabled(bool enabled);
+
+    modifier nonZeroAddress(address addr) {
+        require(addr != address(0), "OperatorManager: zero address");
+        _;
+    }
+
+    constructor(address[] memory _operators, address[] memory _feeReceivers, bool _allowListEnabled) Ownable() {
+        allowListEnabled = _allowListEnabled;
+        bool[] memory _allowed = new bool[](_operators.length);
+        for (uint256 i = 0; i < _operators.length; i++) {
+            _allowed[i] = true;
+        }
+        _setOperators(_operators, _allowed, _feeReceivers);
+    }
+
+    /**
+     * @dev Doesn't return any data, as operator URI is not used in this implementation.
+     */
+    function operatorURI() external pure returns (string memory) {
+        return "";
+    }
+
+    /**
+     * @dev Sets the allow list enabled flag.
+     * @param _allowListEnabled flag to enable or disable allow list.
+     */
+    function setAllowListEnabled(bool _allowListEnabled) external onlyOwner {
+        allowListEnabled = _allowListEnabled;
+        emit UpdateAllowListEnabled(_allowListEnabled);
+    }
+
+    /**
+     * @dev Adds or removes an operator from the allow list.
+     * @param _operator address of the operator.
+     * @param _allowed flag to enable or disable operator.
+     * @param _feeReceiver address of the fee receiver.
+     */
+    function setOperator(address _operator, address _feeReceiver, bool _allowed) external onlyOwner {
+        _setOperator(_operator, _allowed, _feeReceiver);
+    }
+
+    /**
+     * @dev Adds or removes operators from the allow list.
+     * @param _operators addresses of the operators.
+     * @param _allowed flags to enable or disable operators.
+     * @param _feeReceivers addresses of the fee receivers.
+     */
+    function setOperators(
+        address[] calldata _operators,
+        bool[] calldata _allowed,
+        address[] calldata _feeReceivers
+    )
+        external
+        onlyOwner
+    {
+        _setOperators(_operators, _allowed, _feeReceivers);
+    }
+
+    /**
+     * @dev Sets the fee receiver for the operator.
+     * @param _feeReceiver address of the fee receiver.
+     */
+    function setFeeReceiver(address _feeReceiver) external {
+        require(isOperator(msg.sender), "OperatorManager: operator not allowed");
+        operatorFeeReceiver[msg.sender] = _feeReceiver;
+        emit UpdateOperator(msg.sender, _feeReceiver, true);
+    }
+
+    function _setOperators(
+        address[] memory _operators,
+        bool[] memory _allowed,
+        address[] memory _feeReceivers
```

Comment:
I think we can keep `calldata` location here?

Reply:
I've completely removed this function in https://github.com/zkBob/zkbob-contracts/pull/97/commits/ecc926b6341fcbefec73e72c9846b161f270934f. There is a bit of code duplication, but I believe it is still better than before.